web/styles/subscriptions.css

```diff
@@ -792,20 +794,18 @@ h4.user_group_setting_subsection_title {
             margin-left: -3px;
             margin-top: -5px;
             white-space: nowrap;
-            max-width: 260px;
 
             .group-name,
             .sub-stream-name {
                 display: block;
-                max-width: 20ch;
+                max-width: 100%;
                 overflow: hidden;
                 text-overflow: ellipsis;
-                float: left;
             }
         }
 
         .button-group {
-            float: right;
+            margin-left: 1rem;
```

Comment:
Sorry I am bit late to comment but I think this should be `margin-left: auto;` so that edit (pencil) button is automatically aligned towards the right end, and is less fragile to changes.
This change causes the button-group to shift to the end of group names for shorter group names.
 
![image](https://user-images.githubusercontent.com/63504956/222777947-d3bd1f35-bfa2-4199-b2f8-64eaacf1e330.png)

Reply:
@Ddharmani3 Since this PR has been merged, you can put up a new PR with the suggested change.

Reply:
@alya @m-e-l-u-h-a-n Here is the [PR](https://github.com/zulip/zulip/pull/24583) incorporating suggested changes.